Make SSL optional when cert is set for ingestor and kibana

janaurka opened the pull request:
This introduces `properties.haproxy.kibana.enable_ssl` and
`properties.haproxy.ingestor.enable_ssl` (both defaults to false)
which take care of enable SSL on the haproxy more flexible then
before.

Previously if a SSL cert was given, port `5514` (ingestor) and
kibana, port 80 (HTTP) or 443 (HTTPS) was automatically enabled.

With this commit, one can be enabled when the other is disabled.

Example:
Ingestor (syslog) traffic should be plainext, but the
Kibana traffic should be encrypted.

Commit:
Make SSL optional when cert is set for ingestor and kibana

This introduces `properties.haproxy.kibana.enable_ssl` and
`properties.haproxy.ingestor.enable_ssl` (both defaults to false)
which take care of enable SSL on the haproxy more flexible then
before.

Previously if a SSL cert was given, port `5514` (ingestor) and
kibana, port 80 (HTTP) or 443 (HTTPS) was automatically enabled.

With this commit, one can be enabled when the other is disabled.

Example:
Ingestor (syslog) traffic should be plainext, but the
Kibana traffic should be encrypted.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/cloudfoundry-community/logsearch-boshrelease?pullRequest=43) <br/>All committers have signed the CLA.

Infra-Red commented on jobs/haproxy/templates/haproxy.config.erb:
```diff
@@ -86,14 +86,17 @@ backend cluster_monitor
     mode http
 
-    <% if properties.haproxy.ssl_pem %>
+    <% if properties.haproxy.ssl_pem and properties.haproxy.elastic.enable_ssl == true %>
```
This property is missing in haproxy job spec.

Commit:
Add properties.haproxy.elastic.enable_ssl to specfile

The properties.haproxy.elastic.enable_ssl property was missing in
the spec file as @Infra-Red pointed out correctly.

Also the `default` values for `enable_ssl` have been changed to
true to be backward compatible. If an ssl cert is set, ssl will be
enabled on all the services on ha_proxy but can be disabled by
`enable_ssl: false`.

janaurka commented:
@Infra-Red: You are right, should be fixed with 7ab1ecb.

Infra-Red commented on jobs/haproxy/spec:
```diff
@@ -69,6 +75,9 @@ properties:
     description: "Lish of Elasticsearch nodes"
     default: []
+  haproxy.elastic.enable_ssl:
+    description: "Enable or Disable SSL for for syslog/ingestor traffic on haproxy"
```
@janaurka Could you please also update property description?

Commit:
Update descripiton for haproxy.elastic.enable_ssl

janaurka committed:
Merge branch 'develop' into haproxy_ssl

janaurka commented:
@Infra-Red: sorry, this somehow went unnoticed. Should be fixed now.
